app/services-manager.ts

```diff
@@ -407,8 +438,31 @@ export class ServicesManager extends Service {
   }
 
 
+  /**
+   * the information about resource scheme helps to improve performance for API clients
+   * this is undocumented feature is mainly for our API client that we're using in tests
+   */
+  getResourceScheme(resourceId: string): Dictionary<string> {
+    const resource = this.getResource(resourceId);
+    if (!resource) {
+      this.requestErrors.push(`Resource not found: ${resourceId}`);
+      return null;
+    }
+    const resourceScheme = {};
+
+    Object.keys(resource.__proto__).concat(Object.keys(resource))
```

Comment:
`__proto__` is considered a non-standard feature, and `Object.getPrototypeOf` is preferred